driver/avgpool_driver.hpp

```diff
@@ -0,0 +1,595 @@
+
+        float kernel_average_time =
+            iter > 1 ? (kernel_total_time - kernel_first_time) / (iter - 1) : kernel_first_time;
+        printf("GPU Kernel Time Backward AvgPool Elapsed: %f ms\n", kernel_average_time);
+    }
+
+    input_grad_dev->FromGPU(GetStream(), input_grad.data());
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int AvgPoolDriver<Tgpu, Tref>::RunBackwardCPU()
+{
+    if(in_dim.size() == 4)
+    {
+        mloAvgPoolBackward2dRunHost<Tgpu, Tref>(outputGradDesc,
+                                                inputGradDesc,
+                                                output_grad.data(),
+                                                input_grad_host.data(),
+                                                N,
+                                                C,
+                                                H,
+                                                W,
+                                                OH,
+                                                OW,
+                                                ksize.data(),
+                                                stride.data(),
+                                                padding.data(),
+                                                count_include_pad,
+                                                divisor_override);
+    }
+    else if(in_dim.size() == 5)
+    {
+        mloAvgPoolBackward3dRunHost<Tgpu, Tref>(outputGradDesc,
+                                                inputGradDesc,
+                                                output_grad.data(),
+                                                input_grad_host.data(),
+                                                N,
+                                                C,
+                                                D,
+                                                H,
+                                                W,
+                                                OD,
+                                                OH,
+                                                OW,
+                                                ksize.data(),
+                                                stride.data(),
+                                                padding.data(),
+                                                count_include_pad,
+                                                divisor_override);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+Tref AvgPoolDriver<Tgpu, Tref>::GetTolerance()
+{
+    // Computation error of fp16 is ~2^13 (=8192) bigger than
+    // the one of fp32 because mantissa is shorter by 13 bits.
+    auto tolerance = std::is_same<Tgpu, float>::value ? 1.5e-6 : 8.2e-3;
+
+    // bf16 mantissa has 7 bits, by 3 bits shorter than fp16.
+    if(std::is_same<Tgpu, bfloat16>::value)
+        tolerance *= 8.0;
```

Comment:
Is this from AMD's demand?, if not, it is accepted by Moreh's engineer

Reply:
I agree with @cognaiger9. Our recently PRs in ROCm/MIOpen are forced to use `AddTensorFlag` from AMD.

Reply:
> Is this from AMD's demand?, if not, it is accepted by Moreh's engineer

Confirmed that in latest pull request GLU they require to avoid using magic number like this

Reply:
fixed it